### PR TITLE
fix(vue): remove mock facade values code

### PR DESCRIPTION
**Pull request by wlee221**
#### Problem

Previously, `useAuthenticator` relied on a mock facade value object to to infer the shape of `useAuthenticator` return value.

This is problematic, because the object is not strongly typed and manually maintained. For example, it already was missing `socialProviders`, and so it was never picked up in the return value.

#### Solution

This PR uses the return value of `getServiceContextFacade` directly as a source of truth.

#### Description of how you validated changes

Existing e2e tests and unit tests, as well as basic smoke testing:

https://github.com/aws-amplify/amplify-ui/assets/43682783/cd5c18b4-1836-4ce0-bebc-c87eaac28e34

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/vue/src/composables/useAuth.ts
Comment (wlee221) [inline packages/vue/src/composables/useAuth.ts:10]:
```suggestion
```

File: packages/vue/src/composables/useAuth.ts
Comment (wlee221) [inline packages/vue/src/composables/useAuth.ts:72]:
```suggestion
      //@ts-ignore
```

File: packages/vue/src/composables/useAuth.ts
Comment (wlee221) [inline packages/vue/src/composables/useAuth.ts:77]:
```suggestion
    useAuthenticatorValue.state = state;
```

File: packages/vue/src/composables/useUtils.ts
Comment (wlee221) [inline packages/vue/src/composables/useUtils.ts:4]: `socialProvider` was in particular broken because this object didn't have it.

File: packages/vue/src/composables/__tests__/useAuthenticator.spec.ts
Comment (wlee221) [inline packages/vue/src/composables/__tests__/useAuthenticator.spec.ts:63]:
```suggestion
  it('calls getServiceFacade once on initial mount', () => {
```

File: packages/vue/src/composables/useAuth.ts
Comment (wlee221) [inline packages/vue/src/composables/useAuth.ts:60]: This initial value is not needed, because `watchEffect` is guaranteed to run immediately (see comment below). And `...facade` was spreading a mock value anyway, so we aren't losing anything.

File: packages/vue/src/composables/__tests__/useAuthenticator.spec.ts
Comment (wlee221) [inline packages/vue/src/composables/__tests__/useAuthenticator.spec.ts:23]: Added this because `socialProviders` was previously broken.

File: packages/vue/src/composables/useAuth.ts
Comment (wlee221) [inline packages/vue/src/composables/useAuth.ts:56]: This initial value is not needed, because `watchEffect` below is guaranteed to run immediately (see comment below). And `...facade` was spreading a mock value anyway, so we aren't losing anything.

I actually prefer this because now we only have one place where we populate `useAuthenticatorValue`.

**Review by reesscot — APPROVED** [review]
LGTM, please wait till liveness package release goes out before merging

**Comment by wlee221** [comment]
> LGTM, please wait till liveness package release goes out before merging

But I will, it's a feature branch ;)

**Pull request merged**
